lib/bugsnag/integrations/railtie.rb

```diff
@@ -73,9 +74,15 @@ class Railtie < ::Rails::Railtie
     # @api private
     # @param event [Hash] details of the event to subscribe to
     def event_subscription(event)
-      ActiveSupport::Notifications.subscribe(event[:id]) do |*, data|
+      ActiveSupport::Notifications.subscribe(event[:id]) do |*, event_id, data|
         filtered_data = data.slice(*event[:allowed_data])
         filtered_data[:event_name] = event[:id]
+        filtered_data[:event_id] = event_id
+        if event[:id] == "sql.active_record"
+          binds = data[:binds].map { |bind| defined?(bind.name) ? [bind.name, '?'] : nil }
```

Comment:
I don't think `binds` is used when using the mysql2 activerecord adapter, so queries (in the `:sql` entry of the data/payload) won't be redacted for Rails apps using mysql2.

For reference, I was testing a Rails 5.1 app, and something like `Post.where(title: 'foobar')` resulted in the following notification payload:
```
{:sql=>"SELECT  `posts`.* FROM `posts` WHERE `posts`.`title` = 'foobar' LIMIT 11", :name=>"Post Load", :binds=>[], :type_casted_binds=>[], :statement_name=>nil, :connection_id=>70150552673420}
```

Reply:
So long as the query quotes stay the same as in that sql string (which I'll verify), it should be something fairly easily replaceable.